bbb/services.py

```diff
@@ -595,7 +595,13 @@ def handleException(self, data, msg):
             # BuildRequest.
             else:
                 log.info("task %s: buildrequest %s: BuildRequest found for task, cancelling it.", taskid, brid)
-                self.selfserve.cancelBuildRequest(branch, brid)
+                try:
+                    self.selfserve.cancelBuildRequest(branch, brid)
+                except HTTPError as e:
+                    if e.response.status_code == 404:
+                        log.warn("Failed to cancel build request %s on %s. Ignoring the failure.", brid, branch)
```

Comment:
please add the task id and buildrequest prefix here, to make log grepping easier